pkg/controller/containersnapshot/containersnapshot_controller.go

```diff
@@ -179,36 +181,38 @@ func (r *ReconcileContainerSnapshot) onCreation(ctx context.Context, cr *atomv1a
 			})
 		}
 
+		stale = true
+		cr.Status.WorkerState = atomv1alpha1.WorkerFailed
+
 		return reconcile.Result{}, e
 	}
 
 	stale = cr.Status.NodeName != nodeName || cr.Status.ContainerID != containerID
 	cr.Status.NodeName = nodeName
 	cr.Status.ContainerID = containerID
 
-	// Define a new Pod object
-	pod := r.newWorkerPod(cr)
-	// Set ContainerSnapshot instance as the owner and controller
-	if err := controllerutil.SetControllerReference(cr, pod, r.scheme); err != nil {
-		return reconcile.Result{}, err
+	// Check if this Pod already exists
+	pod, e := r.getWorkerPod(ctx, cr.Namespace, cr.UID)
+	if !errors.IsNotFound(e) && !stderr.Is(e, errWorkerPodNotFound) {
+		reqLogger.Info("pod already exists, delete it")
```

Comment:
why delete if already created?

Reply:
before this pr, worker pod is selected by label or owner name, we can not tell if it is created by current snapshot or by an old snapshot with the same name. it should be clear now since we select pod by owner uid, and we do not need to delete it.

Reply:
then why delete here?